### DISCUSSION TITLE
What is the best way to "freeze" parts of a protein?

### DISCUSSION THREAD

**Discussion by MeeMaster** [OpenMD/OpenMD#26] [Q&A]
I tried to use the `rigidBody` field to constrain parts of a protein (e.g. backbone). However, after defining the `rigidBody` field, I got an error:

```
DumpReader Error: Directional StuntDouble 698 has no
	Quaternion Field ("q") specified.
       698      pv             -6.529             -3.672             -6.701  0.000000e+00  0.000000e+00  0.000000e+00
```
This is a rigid body atom that is connecting with a non-rigid body atom. There is no information on how to create quaternion fields. How can I deal with this problem?

The best approach I found that is somewhat working, is to use `restraint` field and apply an extremely high `displacementSpringConstant` on the desired atoms. This however still calculates forces for all of those atoms during simulation.

**Comment by gezelter** [comment]
A rigidBody should have a single line which gives the position of the center of mass of the rigid body, and quaternions describing it's orientation in space.   The quaternion fields require a rotation matrix, A, composed of unit vectors along the principal axes of the rigid body.   If you put only the atoms you want to comprise the rigid body in an xyz file,  you can find the quaternions using the principalAxisCalculator program:

`principalAxisCalculator -x rigidAtoms.xyz -o rotated.xyz`

I think the quaternion fields should be printed out when you include the -o flag.

There may be a way to do what you want with distance constraints internal to the structure, but there would be lots of constraints.  This is not documented well yet, but inside a molecule definition, you can add lines like:

`  constraint { members(12,13); constrainedDistance = 0.945; }`

Which will constrain a distance between atoms 12 and 13.  Structures with too many constraints can be hard to satisfy, however.
